cpp/src/io/orc/dict_enc.cu

```diff
@@ -432,18 +432,15 @@ __global__ void __launch_bounds__(block_size)
  * @param[in] num_columns Number of columns
  * @param[in] num_rowgroups Number of row groups
  * @param[in] stream CUDA stream to use, default 0
- *
- * @return cudaSuccess if successful, a CUDA error code otherwise
- **/
-cudaError_t InitDictionaryIndices(DictionaryChunk *chunks,
-                                  uint32_t num_columns,
-                                  uint32_t num_rowgroups,
-                                  cudaStream_t stream)
+ */
+void InitDictionaryIndices(DictionaryChunk *chunks,
+                           uint32_t num_columns,
+                           uint32_t num_rowgroups,
+                           cudaStream_t stream)
 {
   dim3 dim_block(512, 1);  // 512 threads per chunk
   dim3 dim_grid(num_columns, num_rowgroups);
   gpuInitDictionaryIndices<512><<<dim_grid, dim_block, 0, stream>>>(chunks, num_columns);
```

Comment:
Should functions like this use the `CHECK_CUDA(stream)` macro to catch errors in debug build?

Reply:
I'm not sure about this one. After some kernel launches we have 
`CUDA_TRY(cudaGetLastError())`
Do you recommend using `CUDA_CHECK` instead?

Either way, I would prefer to make this change in a separate PR, since these are not the only functions that are missing this check.

Reply:
Typically we use `CHECK_CUDA(stream)` from `cudf/utilities/error.hpp`.

I actually don't think it is needed after *every* kernel function. More like every cuDF API that performs asynchronous CUDA calls (kernels, memcpyAsync, memsetAsync).